### ISSUE TITLE
508-defect-level-2 [SCREENREADER]: "In this section" mobile menu does not trap VO cursor on iOS 

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/va.gov-team#28709]
### Original Description ###
The VoiceOver cursor is _not_ trapped within the mobile version of the "In this section" menu in Safari on iOS. For users navigating with VoiceOver on their iPhone, this can pose an issue because they can continue navigating outside of the menu without realizing they have left it.  The menu announces "Start of list" correctly, but does not announce "end of list".

Also, menu stays open once you navigate away from it, resulting in the VoiceOver cursor being set on elements that are obscured by the menu.

Note that VoiceOver cursor is different from tab focus. Tab focus seems to be OK in this situation.

I discussed with @joshkimux, he suggested adding `role="dialog" aria-modal="true"` should prevent the VO cursor from jumping out

### Update/Simplified Description ###
The "In the section" sidebar navigation used within the benefits hub should be treated and marked up as a dialog element. The correct behavior should be:

- Keyboard focus should be trapped inside the menu/drawer when it is open
- Menu/Drawer should be able to be closed using the X at the top or the ESC key
  - If the menu is closed (rather than the user going to another link/page) focus is brought back to the trigger button
- Focus should go to the close button when the drawer first opens - This is currently happening but just noting that this should not change
- The entire element should be wrapped using the semantic `<dialog>` tag
  - If the semantic tag cannot be used `role=dialog` can be substituted
  - Add `aria-modal=true` attribute to the dialog
  - Add `aria-labelledby` attribute to the dialog to point to menu title
  - The semantic `<nav>` element would be inside of the dialog

## Acceptance Criteria

- [ ] VoiceOver cursor is trapped within the mobile menu
- [ ] Proper dialog markup is added to the menu drawer
- [ ] Accessibility review

## Environment

- Initially saw in Safari on iPhone 7
- Replicated in Safari on iPhone 12

## Steps to Recreate

1. Use an iOS device
2. Use Safari to go to a page that has the "In this section" menu such as https://www.va.gov/health-care/family-caregiver-benefits/
3. Turn on VoiceOver (Tell Siri to turn it on, or go to Settings > a11y > Voiceover > On)
4. Go back to Safari (double tap = touch/click)
5. Swipe right to navigate through links
6. Press enter when you land on In this section button to activate menu
7. Swipe right through links - notice when you get to the end of the menu, VoiceOver doesn't tell you, and you can continue to navigate through the page.

[Video](https://user-images.githubusercontent.com/55992116/129388431-7a2d1eca-3b4e-4644-b96e-5c8b829df986.mp4)

**Comment by caw310** [comment]
Closing as this is old and the VADS has been updated. If there is still an issue file a new bug.

**Comment by andaleliz** [comment]
@caw310 I just tested this on Safari iOS and this issue is still present.  It doesn't feel like a good use of time to create an entirely new bug with identical information. 

If you feel strongly that's needed, would you please create it and post the number on this ticket so they can be related? This still needs to be resolved.

cc @artsymartha68 for awareness

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/va.gov-team/28709?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @powellkerry @rmessina1010 @rsmithadhoc

**Comment by micahchiang** [comment]
We may own this, since DST now owns header code, which was always part of the plan. We should begin by checking whether or not a fix for this needs to be done the [header app](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/platform/site-wide/header) in vets-website.

**Comment by jilladams** [comment]
Added to our board. @laflannery fysa, defect 2 in the left sidenav on Benefit hubs / detail pages.

**Comment by jilladams** [comment]
@andaleliz this looks like a finding that popped up during a Collab Cycle - did this pop up during usability research?

**Comment by andaleliz** [comment]
I don't think so. IIRC, this was first encountered this during an accessibility audit, separate from the collaboration cycle. I _think_ we were doing some internal testing on mobile with Angela Fowler, who was a specialist pre-CAIA.

**Comment by FranECross** [comment]
@laflannery Please review and groom if necessary.

**Comment by laflannery** [comment]
@FranECross I added a Simplified Description section to include all the markup that should be updated. This is actually quite similar to the new, proposed filter drawer that we have for the search filters so I am matching this pattern to that.

**Comment by chriskim2311** [comment]
This will carry over. Didn't get to it till the last day of sprint.

**Comment by chriskim2311** [comment]
@laflannery Can you verify that this is still happening with the changes in other my PR? 

I have a PR that fixes the accordion drop-downs with this PR: https://github.com/department-of-veterans-affairs/vets-website/pull/33717

Here is the RI: http://b39e16bc1e69090e1df84c3fc6cdb959.review.vetsgov-internal/family-and-caregiver-benefits/health-and-disability/

**Comment by laflannery** [comment]
@chriskim2311 the issue is with VoiceOver and iOS so I am not going to be able to test in a RI. However, the markup still looks the same, it doesn't have dialog markup that is indicated in the description so the issue is most likely still happening. Are we able to change the markup to make this be a dialog?

**Comment by chriskim2311** [comment]
Currently in PR will close early next sprint

**Comment by Shamiaalam** [comment]
@chriskim2311 This issue is still occurring. Both @laflannery  and I have reviewed it, and the focus does not remain trapped within the navigation menu drawer. Is there a specific reason you opted for using the role="dialog" attribute instead of the semantic <dialog> element? With role="dialog", custom focus management is required, whereas the <dialog> element provides built-in focus management. Switching to <dialog> could resolve the issue.

**Comment by laflannery** [comment]
@chriskim2311 I also want to take ownership of spacing on the custom focus when substituting with `role="dialog"` This is completely on me

**Comment by jilladams** [comment]
I'm a little confused about state of this one. I had a note elsewhere that it had shipped and just needed to be verified on Mobile, but the comments above have me wondering. @chriskim2311 could you verify if you still have anything in flight here from Shamia / Laura's comments above?

**Comment by Shamiaalam** [comment]
@jilladams I already discussed this issue with @chriskim2311

**Comment by Shamiaalam** [comment]
@chriskim2311 let me know if you need any help with reviewing!

**Comment by Shamiaalam** [comment]
"I discussed this issue with @chriskim2311 and we found that changing to semantic HTML markup with a <dialog> tag doesn't open the modal. So, the solution here would be to use role="dialog" instead." for instant.

**Comment by chriskim2311** [comment]
Closing with support from @Shamiaalam to use role="dialog"
